Networking/Networking/Remote/MediaRemote.swift

```diff
@@ -3,6 +3,39 @@ import Foundation
 /// Media: Remote Endpoints
 ///
 public class MediaRemote: Remote {
+    /// Loads an array of media from the site's WP Media Library.
+    ///
+    /// - Parameters:
+    ///   - siteID: Site for which we'll load the media from.
+    ///   - pageFirstIndex: The index of the first page from the caller's perspective, which `pageNumber` is based on.
+    ///   - pageNumber: The index of the page of media data to load from.
+    ///   - pageSize: The number of media items to return.
+    ///   - completion: Closure to be executed upon completion.
+    ///
+    public func loadMediaLibrary(for siteID: Int64,
+                                 pageFirstIndex: Int = Constants.pageFirstIndex,
+                                 pageNumber: Int = Constants.pageFirstIndex,
+                                 pageSize: Int = 25,
+                                 context: String? = nil,
+                                 completion: @escaping (_ mediaItems: [Media]?, _ error: Error?) -> Void) {
+        let parameters: [String: Any] = [
+            ParameterKey.contextKey: context ?? Default.context,
+            ParameterKey.perPage: pageSize,
+            ParameterKey.pageNumber: pageNumber - pageFirstIndex + Constants.pageFirstIndex,
```

Comment:
yea the `pageSize`, `pageNumber`, and `pageFirstIndex` params are used for the infinite scroll. I added `pageFirstIndex` so that the API detail of the page index base is in the remote layer, and the caller can have the `pageNumber` based on any index like 1 or 0. Since this is calling a dotcom endpoint instead of WC endpoint, I was hoping to hide the knowledge of paging index base from the app layer. From the [API documentation (`page` field)](https://developer.wordpress.com/docs/api/1.2/get/sites/%24site/media/), the first page index is 1, which is the same as other WC endpoints but it could potentially be different.

on the other hand, I do understand the confusion of the `pageFirstIndex` param. do you think it'd be better to remove this param, and specify the index base of 1 in the `pageNumber` param documentation?

Reply:
> on the other hand, I do understand the confusion of the pageFirstIndex param. do you think it'd be better to remove this param, and specify the index base of 1 in the pageNumber param documentation?

Yes, I believe it's better if we don't have the `pageFirstIndex` param. And yes, it's great if we can add a comment about the base 1 index. 

I did a quick check of the API endpoints and it looks like the `page` parameter is usually base 1. And it doesn't look like we have arguments similar to `pageFirstIndex` in our `*Remote` classes too. So diverging, in this case, might just add more confusion.

Reply:
yup! removed `pageFirstIndex` here: https://github.com/woocommerce/woocommerce-ios/pull/1900/commits/cad78267095d265c2020d652b65f7cb971365923